### PR TITLE
Update BEST_PRACTICES

### PR DESCRIPTION
**Pull request by elland**

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -529,6 +529,18 @@ attendeeList.sort { a, b in
 }
 ```
 
+Always used named parameters, and never unamed ordered parameters (`$0`). It's harder to write, harder to read and takes much longer to compile.
```
Comment (3lvis) [inline BEST_PRACTICES.md:532]: > takes much longer to compile

OMG. Really?

Comment (elland) [reply BEST_PRACTICES.md:532]: That and `array.append`. 😢

Comment (3lvis) [reply BEST_PRACTICES.md:532]: wow. Is it personal experience or you found it somewhere? Would be awesome to add the source.

File: BEST_PRACTICES.md [hunk 2]
```diff
@@ -529,6 +529,18 @@ attendeeList.sort { a, b in
...
+
+**Preferred:**
+```swift
+array.map { i in return i ** i }
+```
+
+**Not Preferred:**
+```swift
+array.map { return $0 ** $0 }
```
Comment (3lvis) [inline BEST_PRACTICES.md]: Could you add a more realistic example? Maybe

``` swift
let values = [2.0,4.0,5.0,7.0]
let squares = values.map {$0 * $0}
// [4.0, 16.0, 25.0, 49.0]
```

File: BEST_PRACTICES.md (hunk 1 above)
Comment (elland) [reply BEST_PRACTICES.md:532]: I linked you an article and a tweet about this earlier...  http://irace.me/swift-profiling

File: BEST_PRACTICES.md (hunk 2 above)
Comment (elland) [reply BEST_PRACTICES.md]: 🆗

File: BEST_PRACTICES.md [hunk 3]
```diff
@@ -533,12 +533,16 @@ Always used named parameters, and never unamed ordered parameters (`$0`). It's h
 
 **Preferred:**
 ```swift
-array.map { i in return i ** i }
+let values = [2.0,4.0,5.0,7.0]
```
Comment (3lvis) [inline BEST_PRACTICES.md]: Some spaces after the commas would be helpful.

File: BEST_PRACTICES.md [hunk 4]
```diff
@@ -533,12 +533,16 @@ Always used named parameters, and never unamed ordered parameters (`$0`). It's h
...
+let squares = values.map { element in element * element}
```
Comment (3lvis) [inline BEST_PRACTICES.md:537]: I would go for `value in value * value` haha, damn, `$0 * $0` is much better for this specific case.

Comment (elland) [reply BEST_PRACTICES.md:537]: No it isn't. 👍

**Comment by 3lvis** [comment]
Would be nice to see a case where `$0 * $0` was worst than the other approach, so far I can only think on cases where using `$0` would be fine.

File: BEST_PRACTICES.md (hunk 3 above)
Comment (elland) [reply BEST_PRACTICES.md]: I copied your code. 😳

File: BEST_PRACTICES.md (hunk 1 above)
Comment (3lvis) [reply BEST_PRACTICES.md:532]: Gonna put it in my Pinboard reading list.

**Comment by 3lvis** [comment]
I'm up for not deviating too much from the Swift community. Pretty much everywhere people use $0 and $1 when possible. I think we should do the same.

**Comment by marijnschilling** [comment]
I agree with Elland on this one, I think it's way easier to read and thus write. I don't care about the swift community, I care about the clarity of our code :v:

**Comment by marijnschilling** [comment]
What would be the downside of writing this different than most of the swift community @Elvis?

**Comment by elland** [comment]
As @marijnschilling said. I will not compromise the clarity of my code just so that it looks like the code from random people I don't know.

**Comment by 3lvis** [comment]
@marijnschilling consistency, also making it easier for people to contribute since they most likely will follow what the rest of the Swift community does, following the "standard" means less "please fix this" comments.

**Comment by elland** [comment]
@3lvis let's not optimise for a problem we don't have.

**Comment by 3lvis** [comment]
Sounds like you have an agreement. I would prefer to keep things aligned to what the rest of the people are doing as much as possible, but if you both agree and want to have it more verbose I think we should go for it. Gonna merge this now.

🚀

**Comment by elland** [comment]
👏

**Pull request merged**